src/hotspot/share/utilities/globalDefinitions.hpp

```diff
@@ -1105,7 +1106,12 @@ template<class T> constexpr T MIN3(T a, T b, T c)      { return MIN2(MIN2(a, b),
 template<class T> constexpr T MAX4(T a, T b, T c, T d) { return MAX2(MAX3(a, b, c), d); }
 template<class T> constexpr T MIN4(T a, T b, T c, T d) { return MIN2(MIN3(a, b, c), d); }
 
-template<class T> inline T ABS(T x)                 { return (x > 0) ? x : -x; }
+template<class T> inline T ABS(T x)                 {
+  assert(x != std::numeric_limits<T>::min(), "ABS: argument should not allow overflow");
```

Comment:
Shouldn't this check for an integral type? This code makes no sense for floating-point types.

Reply:
Yeah, true. Amended in new commit.